crates/nu-std/lib/mod.nu

```diff
@@ -33,13 +33,45 @@ export use dt [datetime-diff, pretty-print-duration]
 # │ 4 │ fooo     │
 # ╰───┴──────────╯
 # ```
+# - adding paths based on the operating system
+# ```nushell
+# >_ std path add {linux: "foo", windows: "bar", darwin: "baz"}
+# ```
 export def-env "path add" [
     --ret (-r)  # return $env.PATH, useful in pipelines to avoid scoping.
     --append (-a)  # append to $env.PATH instead of prepending to.
     ...paths  # the paths to add to $env.PATH.
 ] {
+    let span = (metadata $paths).span
+    let paths = ($paths | flatten)
+
+    if ($paths | is-empty) or ($paths | length) == 0 {
+        error make {msg: $"Empty input", label: {
+            text: $"Provide at least one string or a record",
+            start: $span.start,
+            end: $span.end
+        }}
+    }
+
     let path_name = if "PATH" in $env { "PATH" } else { "Path" }
 
+    let paths = ($paths | each {|p|
+        if ($p | describe) == "string" {
+            $p
+        } else if (($p | describe) | str starts-with "record") {
+            $p | get -i $nu.os-info.name
+        }
+    })
+
+    
+    if null in $paths or ($paths |  is-empty) {
+        error make {msg: $"Empty input", label: {
+            text: $"Received a record, that does not contain a ($nu.os-info.name) key",
+            start: $span.start,
+            end: $span.end
+        }}
+    }   
```

Comment:
i'm wondering what would happen in the `else` branch of the `let paths` :open_mouth: 

my advice would be to merge the two blocks here, i.e. remove the last `if` + `error make` and move it to the currently missing `else` branch of the `let paths`, so that the user would have a better idea of which record is problematic :smirk:

Reply:
and as i'm testing the new command, i think it will be mandatory :scream: 


being on `linux`, if i add `print $paths` before the `if`, i get
```
> use std "path add"
> path add {linux: linux} {bar: foo}
╭───┬───────╮
│ 0 │ linux │
╰───┴───────╯
```
so the second record does not trigger the error...

Reply:
EDIT: it's not really about the missing `else` branch of the `if`, my bad...
it's about the `get -i` returning `null` or something :+1:

Reply:
Hey sorry not sure to understand. Spliting the error checking is in case a list contain records but none contain one for the current os I don't think we can make an error from a let? That would capture the error in the let I think (not tested).
Also the records are flattened, so the check it against the flat record were value is an array, not each args but it could if that's needed

Reply:
yes, i see what you mean :thinking:

Reply:
default the output of `$p | get -i $nu.os-info.name` to the empty string and then testing `'' in $paths or ($paths | is-empty)` appears to work :+1:

Reply:
the span of `...paths` is kinda broken though :eyes:

Reply:
but it's not related to your PR